### PR TITLE
Issue: 874:  Implement a ComposerArtifactCoordinates

### PR DESCRIPTION
**Pull request by jcoelho93**
- Adding a ComposerArtifactCoordinates class

### CODE REVIEW

**Comment by jcoelho93** [comment]
Closes #874

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-composer-layout-provider/src/main/java/org/carlspring/strongbox/artifact/coordinates/ComposerArtifactCoordinates.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+    private static final String VERSION = "version";
+
+    private String vendor;
+
+    private String name;
+
+    private String version;
+
+    public ComposerArtifactCoordinates(String vendor, String name, String version){
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-composer-layout-provider/src/main/java/org/carlspring/strongbox/artifact/coordinates/ComposerArtifactCoordinates.java]: Cosmetic: We keep braces on new lines. Would you mind changing this? Thanks! :)

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-composer-layout-provider/src/main/java/org/carlspring/strongbox/artifact/coordinates/ComposerArtifactCoordinates.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+                                    version));
+
+        this.version = version;
+        setCoordinate(VERSION, this.version);
+    }
+
+    @Override
+    public SemanticVersion getNativeVersion()
+    {
+        try{
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-composer-layout-provider/src/main/java/org/carlspring/strongbox/artifact/coordinates/ComposerArtifactCoordinates.java]: Cosmetic: Would you mind checking our [coding convention](https://strongbox.github.io/developer-guide/coding-convention.html) and applying the necessary changes to the code and your IDE? Thanks! :)

**Comment by carlspring** [comment]
Would you mind signing the ICLA, as described in the [Contributing](https://github.com/strongbox/strongbox/blob/master/CONTRIBUTING.md) page?

Also, please, feel free to join our [chat channel](https://chat.carlspring.org/), if you'd like to learn more about the project and/or like to find out what else you could help with.

**Comment by jcoelho93** [comment]
Reopening pull request agains branch `composer-layout-provider`

**Pull request closed without merging**